### DISCUSSION TITLE
Subscription on model changes doesn't work in production

### DISCUSSION THREAD

**Discussion by Regressor** [NilCoalescing/djangochannelsrestframework#54] [FAQ] [answered]
Hi. I'm trying to upgrade old legacy project from python2.7/django1.11 to recent python/django. 
There was channels_api usage in this project so I decide to replace it by your lib. I replaced reply and receive_json base methods to get same format and made subscribe action for model and all works when start runserver command - client gets notification messages on model changes.

But in production we have nginx -> gunicorn + daphne running on a number of servers with common database and redis. So when gunicorn code (even on same server) change object, daphne doesn't see signals and doesn't send notifications. There was no such problems with old channels_api library.

Is there any way to get this work in distributed environment ?

**Comment by hishnash** [comment, marked as answer]
when you deploy in production (the gunicorn HTTP code) does that path end up importing your consumers?

You might need to put an inport statement someway in your application to ensure the consumers are imported when you run the regular HTTP endpoints through gunicorn.

**Reply by Regressor** [reply]
You are right! Added import in gunicorn controlled code and now notifications works. Thank you!
